contracts/ERCBlob.sol

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IERC5018.sol";
+import "./IERCBlob.sol";
+
+interface EthStorageContract {
+    function putBlob(bytes32 key, uint256 blobIdx, uint256 length) external payable;
+
+    function get(bytes32 key, uint256 off, uint256 len) external view returns (bytes memory);
+
+    function remove(bytes32 key) external;
+
+    function upfrontPayment() external view returns (uint256);
+}
+
+contract ERCBlob is IERC5018, IERCBlob {
```

Comment:
To avoid empty function for IERC5018, we may want to include all the interfaces in IERC5018ForBlob